### PR TITLE
Fix TensorFlow and PyTorch compatibility

### PR DESCRIPTION
**Pull request by pcmoritz**
## Related issue number

https://github.com/ray-project/ray/issues/3520

### CODE REVIEW

File: cmake/Modules/BoostExternalProject.cmake [hunk 1]
```diff
@@ -27,14 +27,15 @@ else()
...
   set(Boost_SYSTEM_LIBRARY ${Boost_LIBRARY_DIR}/${CMAKE_STATIC_LIBRARY_PREFIX}boost_system${CMAKE_STATIC_LIBRARY_SUFFIX})
+  set(Boost_THREAD_LIBRARY ${Boost_LIBRARY_DIR}/${CMAKE_STATIC_LIBRARY_PREFIX}boost_thread${CMAKE_STATIC_LIBRARY_SUFFIX})
   set(Boost_FILESYSTEM_LIBRARY ${Boost_LIBRARY_DIR}/${CMAKE_STATIC_LIBRARY_PREFIX}boost_filesystem${CMAKE_STATIC_LIBRARY_SUFFIX})
 
   #set(boost_URL https://github.com/boostorg/boost.git)
-  #set(boost_TAG boost-1.65.1)
+  #set(boost_TAG boost-1.69.0)
 
-  set(Boost_TAR_GZ_URL http://dl.bintray.com/boostorg/release/1.65.1/source/boost_1_65_1.tar.gz)
+  set(Boost_TAR_GZ_URL http://dl.bintray.com/boostorg/release/1.69.0/source/boost_1_69_0.tar.gz)
```
Comment (pcmoritz) [inline cmake/Modules/BoostExternalProject.cmake]: Upgrading boost was needed to fix some incompatibility with macOS

**Comment by pcmoritz** [comment]
jenkins retest this please

**Comment by pcmoritz** [comment]
jenkins retest this please

File: python/README-building-wheels.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ produce .whl files owned by root.
 Inside the root directory (i.e., one level above this python directory), run
 
 ```
-docker run --rm -w /ray -v `pwd`:/ray -ti quay.io/xhochy/arrow_manylinux1_x86_64_base:latest /ray/python/build-wheel-manylinux1.sh
+docker run --rm -w /ray -v `pwd`:/ray -ti rayproject/ray_arrow_manylinux1_x86_64_base:latest /ray/python/build-wheel-manylinux1.sh
```
[the remark this answers is not in the document]
Comment (pcmoritz) [reply python/README-building-wheels.md]: I think this is not needed any more, I'm trying to remove it.

File: cmake/Modules/ArrowExternalProject.cmake [hunk 3]
```diff
@@ -14,11 +14,11 @@
 #  - PLASMA_STATIC_LIB
 #  - PLASMA_SHARED_LIB
 
-set(arrow_URL https://github.com/apache/arrow.git)
-# The PR for this commit is https://github.com/apache/arrow/pull/3154. We
+set(arrow_URL https://github.com/ray-project/arrow.git)
```
Comment (robertnishihara) [inline cmake/Modules/ArrowExternalProject.cmake]: Can you document why we're cloning a fork and maybe include a link to the discussion in the arrow PR?

File: cmake/Modules/ThirdpartyToolchain.cmake [hunk 4]
```diff
@@ -54,11 +54,14 @@ ADD_THIRDPARTY_LIB(boost_system
   STATIC_LIB ${Boost_SYSTEM_LIBRARY})
 ADD_THIRDPARTY_LIB(boost_filesystem
   STATIC_LIB ${Boost_FILESYSTEM_LIBRARY})
+  ADD_THIRDPARTY_LIB(boost_thread
```
Comment (robertnishihara) [inline cmake/Modules/ThirdpartyToolchain.cmake]: fix indentation, same with above lines I think

**Comment by robertnishihara** [comment]
Looks good to me other than the comments I left and the comments discussed in person (that the diff from the main arrow repository should be squashed into a single commit).

**Review by atumanov — APPROVED** [review]

**Pull request merged**